app.go

```diff
@@ -536,3 +567,106 @@ func executeCancelSetSell(cmd command) bool {
 	}
 	return true
 }
+
+func executeSetBuyTrigger(cmd command) bool {
+	//Check that a sell amount exists in the store
+
+	userID := cmd.UserID
+	stock := cmd.Args[0]
+	strAmount := cmd.Args[1]
+
+	account := accountStore.GetAccount(userID);
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+
+	stockTriggerCost, err := currency.NewFromString(strAmount)
+
+	if err != nil {
+		log.Error("Failed to parse currency")
+		return false
+	}
+
+	userAutorequest, err := autoBuyRequestStore.GetAutorequest(stock, userID)
+
+	if err != nil {
+		log.Info("User %s does not have an auto request pending")
+		return false
+	}
+
+	stockTotalValue := userAutorequest.Amount
+
+	wholeShares, cashRemainder := stockTriggerCost.FitsInto(stockTotalValue)
+
+	if wholeShares == 0 {
+		log.Notice("Amount specified to buy less than single stock unit")
+		return true
+	}
+
+	log.Infof("User %s set purchase order for %d shares of stock %s", cmd.UserID, wholeShares, stock)
+
+	// Remove the funds from user now to prevent double spending
+	stockTotalValue.Sub(cashRemainder)
+	account.RemoveFunds(stockTotalValue)
+
+	userAutorequest.Trigger = stockTriggerCost
+
+	// Add stocks to user portfolio
+
+	return true
+}
+
+
+func executeSetSellTrigger(cmd command) bool {
+	//Check that a sell amount exists in the store
+	userID := cmd.UserID
+	stock := cmd.Args[0]
+	strAmount := cmd.Args[1]
+
+	account := accountStore.GetAccount(userID);
+
+	if account == nil {
+		log.Infof("User %s does not have an account", userID)
+		return false
+	}
+
+	stockTriggerCost, err := currency.NewFromString(strAmount)
+
+	if err != nil {
+		log.Error("Failed to parse currency")
+		return false
+	}
+
+
+	userAutorequest, err := autoSellRequestStore.GetAutorequest(stock, userID)
+
+	if err != nil {
+		log.Info("User %s does not have an auto request pending")
+		return false
+	}
+
+	stockTotalValue := userAutorequest.Amount
+
+	wholeShares, cashRemainder := stockTriggerCost.FitsInto(stockTotalValue)
+
+	if wholeShares == 0 {
+		log.Notice("Amount specified to buy less than single stock unit")
+		return true
+	}
+
+	log.Infof("User %s set purchase order for %d shares of stock %s", cmd.UserID, wholeShares, stock)
+
+	// Remove the funds from user now to prevent double spending
+	stockTotalValue.Sub(cashRemainder)
+	account.AddFunds(stockTotalValue)
```

Comment:
What (should) happen to `cashRemainder`? Does it go back to the user's account? Stay in the trigger? Disappear into the ether?